lib/msgraph/paginated.go

```diff
@@ -45,26 +91,41 @@ func iterateSimple[T any](c *Client, ctx context.Context, endpoint string, f fun
 			}
 		}
 		return true
-	})
+	}, queryOpts...)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 	return trace.Wrap(itErr)
 }
 
 // iterate implements pagination for "list" endpoints.
-func (c *Client) iterate(ctx context.Context, endpoint string, query url.Values, header map[string]string, f func(json.RawMessage) bool) error {
+func (c *Client) iterate(ctx context.Context, endpoint string, f func(json.RawMessage) bool, queryOpts ...IterateOpt) error {
+	ic := c.newIterateConfig()
+	for _, opt := range queryOpts {
+		opt(ic)
+	}
+
 	uri := *c.baseURL
 	uri.Path = path.Join(uri.Path, endpoint)
-	pageSize := strconv.Itoa(c.pageSize)
-	if query == nil {
-		query = make(url.Values)
+
+	query := uri.Query()
+	if ic.Select != "" {
+		query.Set("$select", ic.Select)
+	}
+	if ic.Filter != "" {
+		query.Set("$filter", ic.Filter)
+	}
+	if ic.Top > 0 {
+		query.Set("$top", strconv.Itoa(ic.Top))
+	}
+	if ic.Count {
+		query.Set("$count", "true")
```

Comment:
I should probably implement a separate `query` method for `IterateConfig` [like I did in `e/lib/intune/api`](https://github.com/gravitational/teleport.e/blob/31e1a8e9ebeb23b654472cf71c6c292205e0be82/lib/intune/api/inventory.go#L30-L51).